src/elm/Page/Dashboard/Analysis.elm

```diff
@@ -566,9 +584,32 @@ update msg model loggedIn =
                 |> UR.init
                 |> UR.addCmd (fetchAnalysis loggedIn initFilter Nothing)
 
+        ToggleSorting ->
+            let
+                oldFilters =
+                    model.filters
+
+                sortDirection =
+                    if model.filters.direction == ASC then
+                        DESC
+
+                    else
+                        ASC
+
+                newModel =
+                    { model
+                        | filters = { oldFilters | direction = sortDirection }
+                        , reloadOnNextQuery = True
+                        , status = Loading
+                    }
+            in
+            newModel
+                |> UR.init
+                |> UR.addCmd (fetchAnalysis loggedIn newModel.filters Nothing)
```

Comment:
Is there a reason we can't just `List.reverse` the claims? That would mean less loading times and better UX, and I don't think we get other benefits from fetching everything again, other than maybe fetching new claims.

Of course, if we ever need to sort by some other parameters, we could create a function like
```elm
applySort : FilterDirection -> List Claim.Model -> List Claim.Model
```
but I guess that's premature optimization right now :sweat_smile:

Reply:
yeah, the problem is that this is a query that is paginated. Imagine we have 100 claims, the older one from December 2019 and the newest March 2021. If we did the reversing on the frontend like you described we would only reverse the first batch we queried (today is 16). To see easily the claim from 2019 we would need to click the "more" button a couple of times until we had all the claims loaded and then reverse it.

This implementation is done by the backend to allow validators to catch on with old claims that simply were left behind. Some users do like 10, 15 claims each days, so its easy to some to get behind. 

that's why its an API call to reverse, instead of a local one

In the shop the picture is different because there we dump the whole thing

Reply:
Makes sense, I didn't take into account that this was paginated